catalogue/src/main/kotlin/com/tidal/sdk/catalogue/utils/RetrofitProvider.kt

```diff
@@ -0,0 +1,47 @@
+package com.tidal.sdk.catalogue.utils
+
+import com.tidal.sdk.auth.CredentialsProvider
+import com.tidal.sdk.catalogue.generated.infrastructure.Serializer.kotlinxSerializationJson
+import com.tidal.sdk.catalogue.networking.AuthInterceptor
+import com.tidal.sdk.catalogue.networking.DefaultAuthenticator
+import com.tidal.sdk.common.d
+import com.tidal.sdk.common.logger
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Converter
+import retrofit2.Retrofit
+import retrofit2.converter.kotlinx.serialization.asConverterFactory
+import retrofit2.converter.scalars.ScalarsConverterFactory
+
+class RetrofitProvider {
```

Comment:
logaf low: this could just be 1 internal method instead of being a class

Reply:
I'm undecided about this one's final form. I'll keep this in mind however (thought of it myself too, but then decided for this version - for now).